src/Microsoft.AspNetCore.Razor.Design/build/netstandard2.0/Microsoft.AspNetCore.Razor.Design.targets

```diff
@@ -142,14 +155,22 @@
     </ItemGroup>
   </Target>
 
-  <Target Name="_RazorAddBuiltProjectOutputGroupOutput" Condition="'$(RazorCompileOnBuild)' == 'true'" BeforeTargets="BuiltProjectOutputGroup">
-    <ItemGroup>
+  <Target 
+    Name="_RazorAddBuiltProjectOutputGroupOutput"
+    DependsOnTargets="RazorResolveGenerateInputs"
+    BeforeTargets="BuiltProjectOutputGroup">
+
+    <ItemGroup Condition="'@(RazorGenerate)'!= '' and '$(RazorCompileOnBuild)' == 'true'">
```

Comment:
This condition needs to be on the itemgroup so that it will be evaled after the DependsOnTargets